nixos/modules/system/boot/timesyncd.nix

```diff
@@ -46,6 +46,28 @@ with lib;
       wantedBy = [ "sysinit.target" ];
       aliases = [ "dbus-org.freedesktop.timesync1.service" ];
       restartTriggers = [ config.environment.etc."systemd/timesyncd.conf".source ];
+
+      preStart = (
+        # Ensure that we have some stored time to prevent
+        # systemd-timesyncd to resort back to the fallback time.  If
+        # the file doesn't exist we assume that our current system
+        # clock is good enough to provide an initial value.
+        ''
+          if ! [ -f /var/lib/systemd/timesync/clock ]; then
+            test -d /var/lib/systemd/timesync || mkdir -p /var/lib/systemd/timesync
+            touch /var/lib/systemd/timesync/clock
+          fi
```

Comment:
This could be simplified as

```bash
mkdir -p /var/lib/systemd/timesync
if ! [[ -f /var/lib/systemd/timesync/clock ]]; then
    touch /var/lib/systemd/timesync/clock
fi
```

`mkdir -p` already checks if the directory exists before attempting to create it, so there's no need to do that separately.